### ISSUE TITLE
IDE2001 is not compatible with brace placement at the end of a line

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#72189]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/IDE2001-analyzer-incorrectly-triggers/10591708)._

---
In Tools -Options - Text Editor - C# - Code Style, setting "Allow embedded statements on same line" to "No" causes analyzer errors to be reported on every non-embedded statement

---
#### Feedback Bot on 2/20/2024, 04:32 AM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by sharwell** [comment]
IDE2001 is reported for the braces at the end of lines, even when `csharp_new_line_before_open_brace` omits `control_blocks`:

```cs
if (a) {
} else {
}
```

**Comment by CyrusNajmabadi** [comment]
By design.  These are strictly not compatible options.  The new-experimental options are not meant to compliment the existing options.  They're meant to supersede (they do not look at, or otherwise consider *any* other options at all).  In this case, the options is strict and intentional.  Embedded states are simply not allowed to start on the prior line.

**Comment by sharwell** [comment]
As a former user of brace placement at the end of a line who also disallowed embedded statements on the same line, I'm reopening this issue. It's a fairly simple approach to make these work together (only the brace itself would need to be checked differently when the option is set).

**Comment by CyrusNajmabadi** [comment]
Closing out.  If you want this done because it meets a use case of yours, please contribute the fix.  otherwise this just clogs the backlog with stuff that won't happen.
